lib/Analysis/Utility.cpp

```diff
@@ -317,8 +317,9 @@ unsigned ScanLoweringHelper::getAxisBlockStride() {
   for (unsigned dim : order) {
     if (dim == getAxis())
       return stride;
-    stride *= type.getShape()[dim] /
-              (sizePerThreads[dim] * threadsPerWarp[dim] * warpsPerCTA[dim]);
+    stride *= std::max<unsigned int>(
+        1, type.getShape()[dim] /
+               (sizePerThreads[dim] * threadsPerWarp[dim] * warpsPerCTA[dim]));
```

Comment:
should this be a `ceil()` instead

Reply:
It's necessary to use `ceil` if we need to support shapes and layouts that are not power of 2. So do we?

Reply:
I see. Can we do `ceil` anyway as this sounds more generic and probably easier for people to understand. (very much a nit)